src/chum.cpp

```diff
@@ -181,7 +182,7 @@ void Chum::refreshDetails() {
   setStatus(qtTrId("chum-get-package-details"));
 
   QStringList packages;
-  for (const ChumPackage *p: m_packages.values())
+  for (const ChumPackage *p: std::as_const(m_packages))
```

Comment:
`m_packages` is a QHash and I am surprised that such iterator would work. Shouldn't it iterate via pairs of key,value?  As there are many extra files in this PR, I didn't check whether it compiles...

For clarity of reading, I would prefer older version where it is clear that iteration is done through values

Reply:
Yes it does, and is the recommended way of iterating the hash values (not the keys) See https://doc.qt.io/qt-5/qhash.html#details

Reply:
Really odd, as I see example

```C++
QHash<QString, int>::const_iterator i = hash.constBegin();
while (i != hash.constEnd()) {
    cout << i.key() << ": " << i.value() << Qt::endl;
    ++i;
}
```

which means that you cannot assign iterator to `const ChumPackage *p`, but have to get the package via `i.value()`. Something is odd over here, I would have to try to compile whole thing

Reply:
Its the last example:
If you only need to extract the values from a hash (not the keys), you can also use foreach:

QHash<QString, int> hash;
...
foreach (int value, hash)
    cout << value << Qt::endl;

Reply:
But we don't use `foreach`. I was sure that such `for` loop form corresponds to iteration using begin() and end(). So, it is rather confusing, I must say

Reply:
Experimented a bit and, indeed, such iteration goes through values and not the iterator. I guess we can then just make this change. Although, without as_const